bugbug/models/assignee.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from collections import Counter
+
+import xgboost
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features
+from bugbug import bugzilla
+from bugbug.model import Model
+
+
+class AssigneeModel(Model):
+    def __init__(self, lemmatization=False):
+        Model.__init__(self, lemmatization)
+
+        self.cross_validation_enabled = False
+        self.calculate_importance = False
+
+        feature_extractors = [
+            bug_features.has_str(),
+            bug_features.severity(),
+            bug_features.keywords(),
+            bug_features.is_coverity_issue(),
+            bug_features.has_crash_signature(),
+            bug_features.has_url(),
+            bug_features.has_w3c_url(),
+            bug_features.has_github_url(),
+            bug_features.whiteboard(),
+            bug_features.patches(),
+            bug_features.landings(),
+            bug_features.title(),
+        ]
+
+        cleanup_functions = [
+            bug_features.cleanup_fileref,
+            bug_features.cleanup_url,
+            bug_features.cleanup_synonyms,
+        ]
+
+        self.extraction_pipeline = Pipeline([
+            ('bug_extractor', bug_features.BugExtractor(feature_extractors, cleanup_functions, rollback=True)),
```

Comment:
We should probably rollback to the point where the assignee was decided, rather than the point where the bug was filed, but we can do this in a follow-up.

Reply:
What I understand is that, I will need to create a function func_name which will be passed as an argument [rollback_when](https://github.com/mozilla/bugbug/blob/master/bugbug/bug_features.py#L361) to [BugExtractor](https://github.com/mozilla/bugbug/blob/master/bugbug/bug_features.py#L356) and also set rollback=True. 
```python
def func_name(change):
    addresses_to_exclude = [
            'nobody@bugzilla.org',
            'nobody@example.com',
            'nobody@fedoraproject.org',
            'nobody@mozilla.org',
            'nobody@msg1.fake',
            'nobody@nss.bugs',
            'nobody@t4b.me'
    ]
    if change['field_name'] == 'assigned_to':
        if change['removed'] in addresses_to_exclude:
            return(True)
    return(False)

```
func_name() takes an argument [change](https://github.com/mozilla/bugbug/blob/master/bugbug/bug_snapshot.py#L101) from [bug_snapshot.rollback](https://github.com/mozilla/bugbug/blob/master/bugbug/bug_snapshot.py#L96), and returns True if change has "field_name":"assigned_to" and change['removed'] belongs to the addresses associated with "No Assignee".
Is this correct?

Reply:
Yes, correct! But you can simplify by just checking if the assignee was ever changed, you don't need to check that it was changed from a nobody to a non-nobody.

Reply:
As I said though, this can be done in a follow-up! (follow-up means another PR)

Reply:
I will create another PR for this. 
1. Could you suggest where to define 'func_name'?
2. Could you also suggest an appropriate name for 'func_name'? is 'is_change_assignee' a good name?
3. I will follow your advise, however, just for my reference, I wanted to know why wouldn't it be better to rollback to the earliest assignee change from nobody to Non-nobody, as that would preserve all further assignee changes. On the contrary, this will lead to loss of that info.

Reply:
1. You can define it in the model class, look at the other models using rollback.
2. As in the other models, you can just call it `rollback`.
3. What I suggested is just a simplification, the result will be the same (you'll rollback to the first assignee change in both cases)

Reply:
Thanks for the explanation on point 3. I just realized that it will be the case.
Is there anything else left to do on this issue?
I have added the rollback in [another branch](https://github.com/LakshyAAAgrawal/bugbug/tree/rollback_assignee_change)( branched out from this branch ). Should I create a pull request from that, or wait for this to merge?

Reply:
I've merged the PR now.

In general, feel free to open PRs even if they depend on another PR to land first (you can just rebase it once the first lands).